### PR TITLE
Added core Jdex skeleton

### PR DESCRIPTION
**Pull request by TryExceptElse**
created JDex main object method skeletons, and placeholder objects for front and back ends I'm pulling this in from the frontend branch, but it's really an update to the project core.

### CODE REVIEW

**Comment by TryExceptElse** [comment]
Updated CI JDK to 8, removed appWindow class (I'll re-add this later) and all tests are passing.

**Comment by Entrer1** [comment]
I don't see how to make comments on the code itself, but I see the classes that I'm going to write in the future, so that's good!

File: src/main/java/io/github/tryexceptelse/jdex/be/Dex.java [hunk 1]
```diff
@@ -1,19 +1,7 @@
...
-    ArrayList getList();
-
-    Contact addContact();
-
-    ArrayList search(String first, String last, String email, String streetAddr, String phone);
-
-    void save(String address);
-
-    void load(String address);
+public class Dex {
```
Comment (TryExceptElse) [inline src/main/java/io/github/tryexceptelse/jdex/be/Dex.java:6]: This is a placeholder class to be replaced later with a class implementing the IDex interface

**Comment by TryExceptElse** [comment]
click files changed, and you can select a line of code, hit the plus sign, and write comments

**Pull request merged**